### PR TITLE
Clamp negative rle_loss to zero in Pose26 training

### LINKED ISSUE #23577: Maybe rle_loss should set to zero if it become to negative with YOLO26 Pose26 head
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Once the rle_loss become negative, the result will be bad. Maybe rle_loss should set to zero if it become to negative with YOLO26 Pose26 head.

### Environment

latest

### Minimal Reproducible Example

train

### Additional

Let me record my experiments here:

Yesterday, I try to train with the custom dataset from scratch using a modified yaml.

1. When my coco train2017.txt filelines not startswith "./“ it just corrupted because it still use local filepath. So fixed it by this https://github.com/ultralytics/ultralytics/pull/23562

2. As there are 30W(three hundred thousand) images in training dataset, it cost about 2~3 hours to train one epoch. But the the first epoch failed to save, just because the `core` module which used to read result csv to save into `pt` was not imported (not catched by silly try-catch) which I think it's a bug https://github.com/ultralytics/ultralytics/issues/23564. 

https://github.com/ultralytics/ultralytics/blob/1da4e933181f0dfe983349c2b1dcda81dba094f8/ultralytics/engine/trainer.py#L576-L583

3. Then I restart training again in last night.

It still occurs when  training a custum dataset as reported by https://github.com/ultralytics/ultralytics/issues/23233 . Once rle_loss become negative, the result became worse.

```
      1/100      5.12G      1.493      1.299      0.401      5.298    0.01023      15.36         38        960: 90% ━━━━━━━━━━╸─ 18494/20478 1.7it/s 3:23:48<19:0
      1/100      5.14G       1.45      1.187     0.3865      4.836   0.009841      13.72         16        960: 100% ━━━━━━━━━━━━ 20478/20478 1.5it/s 3:42:40
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 4.1it/s 4:40
                   all      36414      36414      0.966      0.622      0.649      0.527      0.966      0.622      0.649      0.648

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
      2/100      5.97G      1.017     0.1307     0.1886     0.4427   0.005956     -1.988         31        960: 56% ━━━━━━╸───── 11438/20478 1.8it/s
      2/100      5.98G     0.9972     0.1199     0.1623     0.4172    0.00582     -2.199         12        960: 100% ━━━━━━━━━━━━ 20478/20478 1.8it/s 3:06:41
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 4.2it/s 4:31
                   all      36414      36414      0.982      0.631      0.672      0.534      0.982      0.631      0.672      0.638

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
      3/100      5.98G     0.9535    0.09833     0.1041     0.3646   0.005546     -2.703          7        960: 100% ━━━━━━━━━━━━ 20478/20478 2.0it/s 2:54:03
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 4.3it/s 4:27
                   all      36414      36414      0.942      0.594      0.637      0.425      0.942      0.594      0.637      0.515

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
      4/100      5.98G     0.9212    0.08306    0.07715     0.3429   0.005332     -3.149         17        960: 100% ━━━━━━━━━━━━ 20478/20478 2.0it/s 2:49:18
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 4.2it/s 4:30
                   all      36414      36414      0.359      0.622      0.574      0.336      0.359      0.623      0.575      0.445

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
      5/100      5.98G     0.8957    0.07515    0.06326     0.3259   0.005168     -3.439         26        960: 73% ━━━━━━━━╸─── 14860/20478 2.2it/s 2:04:22<42:
      5/100      5.98G      0.893    0.07412    0.06224     0.3242   0.005152     -3.471         12        960: 100% ━━━━━━━━━━━━ 20478/20478 2.0it/s 2:51:42
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 4.2it/s 4:32
                   all      36414      36414      0.982      0.463      0.623      0.398      0.982      0.463      0.623      0.541

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
      6/100      5.98G     0.8862    0.07087    0.05777     0.3169     0.0051     -3.578         33        960: 22% ━━╸───────── 4507/20478 2.2it/s
      6/100      5.98G      0.884    0.07048    0.05761     0.3165   0.005086     -3.593         30        960: 35% ━━━━──────── 7121/20478 2.1it/s 59:47<1:45:06
```

The first epoch looks ok.  But latter epochs are not satisfied, the MAP drop too many from 0.648 to 0.541 at 5th epoch, may be just because the `rle_loss` become negtive.
The GPT says the rle_loss  maybe negative when sigma is small and log_phi is large, but I'm not really understand.

4. Just set rle_loss to zero once it become negative, and train again. 

https://github.com/ultralytics/ultralytics/issues/23233#issuecomment-3857463733

> It looks like won't predict keypoints, if I change rle_loss to const zero once if became to negative and then continue training.

This is an error conclusion, after try  again and again.

I create a yolo by `yolo = YOLO("my_pose.pt")`, then I check yolo type, it's `detect`. It will not output keypoints even if `predict(..., task="pose")`, unless set `yolo.task="pose"`.

5. Retrain with my fist epoch ptfile, now the loss looks like normal.

```
      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size                                                                      
       1/30      2.35G      1.084     0.1642     0.1432     0.5401    0.00625          0         37        960: 49% ━━━━━────── 10046/20478 2.1it/s 1:20:46<1:21:52 
       1/30      2.37G      1.054     0.1526    0.09992     0.4969   0.006093          0          6        960: 100% ━━━━━━━━━━━━ 20478/20478 2.4it/s 2:24:27  
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Pose(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1138/1138 5.5it/s 3:28
                   all      36414      36414      0.979      0.639      0.688      0.561      0.979      0.639      0.688      0.687
      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss   rle_loss  Instances       Size
       2/30      2.75G      1.031     0.1407    0.04583      0.439   0.005968          0         33        960: 1% ─────────── 193/20478 3.4it/s 1:06<1:40:22
       2/30      2.75G      1.016     0.1361    0.04311     0.4262   0.005907          0         31        960: 32% ━━━──────── 6551/20478 3.6it/s 38:17<1:03:53
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Summary

Fixes #23577

The `rle_loss` in the YOLO26 Pose head uses a log-likelihood formulation (`log_sigma - log_phi + residual terms`) that can produce increasingly negative values during training. As reported in the issue, this causes the negative loss to dominate the optimization objective, leading to progressive mAP degradation across epochs (e.g., Pose mAP50-95 dropping from 0.648 to 0.445 over just a few epochs).

This PR adds a `rle_loss.clamp(min=0)` call immediately after the RLE loss computation in `calculate_keypoints_loss`. This prevents the negative loss from pulling the total loss downward and destabilizing training, while preserving the normal gradient signal when the loss is positive.

The issue reporter confirmed that clamping negative `rle_loss` to zero stabilized training and produced consistently improving metrics across epochs, with Pose mAP50-95 reaching 0.957 by epoch 14.

## Changes

- **`ultralytics/utils/loss.py`**: Added `rle_loss = rle_loss.clamp(min=0)` after `calculate_rle_loss()` in the `calculate_keypoints_loss` method of `v8PoseLoss`.

## Test plan

- [x] Verified the change is a single-line addition that only affects the RLE loss path (gated behind `self.rle_loss is not None`)
- [ ] Train a YOLO26 Pose model and confirm `rle_loss` no longer goes negative
- [ ] Verify Pose mAP metrics improve consistently across epochs without degradation

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fix Pose26 training stability by clamping negative `rle_loss` values to zero ✅

### 📊 Key Changes
-Added `rle_loss = rle_loss.clamp(min=0)` after RLE loss computation in `ultralytics/utils/loss.py`.
-Applies only when RLE loss is enabled and keypoint prediction format includes RLE params (last dim 4 or 5).

### 🎯 Purpose & Impact
-Prevents unexpected negative RLE loss values from propagating into the total pose loss, improving training stability 🛡️
-Reduces risk of destabilized gradients/logging confusion during Pose26 keypoint training runs 📉
-Minimal, low-risk change localized to pose loss computation with no impact on non-pose tasks 🎯

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
I have read the CLA Document and I sign the CLA

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Mr-Neutr0n, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist (this is automated, and an engineer will assist soon) 🛠️

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

PR looks clean and focused. The clamp is localized, low-risk, and should improve stability without affecting other paths. No issues found.

**Comment by Laughing-q** [comment]
@Mr-Neutr0n nice improvement! merging! thank you!

**Review by Laughing-q — APPROVED** [review]

**Comment by Mr-Neutr0n** [comment]
happy to contribute:)

**Comment by glenn-jocher** [comment]
Thanks for the contribution—since this PR is closed now, if you still see Pose RLE instability on current `main`, please open a new issue with a minimal repro and training logs so we can track it.
